komfydb/execution/hash_join.h

```diff
@@ -0,0 +1,48 @@
+#ifndef __HASH_JOIN_H__
+#define __HASH_JOIN_H__
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include "komfydb/execution/join.h"
+#include "komfydb/execution/join_predicate.h"
+#include "komfydb/storage/record_id.h"
+
+namespace {
+
+using komfydb::storage::RecordId;
+
+};
+
+namespace komfydb::execution {
+
+using RecordVector = std::vector<std::unique_ptr<Record>>;
+
+class HashJoin : public Join {
```

Comment:
I'm wondering if this inheritance is needed here? Do we really need any logic from a loop join here? And shouldn't we rename the Join to LoopJoin actually? (The maybe in the JoinOptimizer the `InstatiateJoin` method should just return a pointer to OpIterator)